src/main/java/org/ice4j/ice/Agent.java

```diff
@@ -114,6 +115,36 @@
     public static final String PROPERTY_ICE_PROCESSING_STATE
                                             = "IceProcessingState";
 
+    /**
+     *  The ScheduledExecutorService to execute planned agent termination
+     */
+    private static final ScheduledExecutorService agentTerminationScheduler;
+
+    static {
+        final ScheduledThreadPoolExecutor terminationExecutor =
+            new ScheduledThreadPoolExecutor(0);
```

Comment:
I'm reading [here] that 0 is not a good idea:
>Additionally, it is almost never a good idea to set corePoolSize to zero or use allowCoreThreadTimeOut because this may leave the pool without threads to handle tasks once they become eligible to run.
I think it would be reasonable to allow more than one termination task to execute at the same time. @bgrozev WDYT ?
[here]: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledThreadPoolExecutor.html#ScheduledThreadPoolExecutor(int)

Reply:
@paweldomas I've also read that. But it is occurred to be inconsistency in doc... I'll find prooflink and attach. I've checked both in test application and inside JVB that it works as expected - create threads on demand and release them after timeout of no work.

Reply:
@paweldomas here is a link https://bugs.java.com/bugdatabase/view_bug.do?bug_id=7091003

Reply:
>I think it would be reasonable to allow more than one termination task to execute at the same time. 

@paweldomas this scheduled thread pool is not limited to one thread (it actually has no limit on threads to create), so many termination tasks from different agents could be executed in parallel. But using single shared thread pool between all agents results in using as low number of threads as possible.

Reply:
> So with pool size 0 there will be no threads kept alive, but does it allow to execute multiple tasks at the same time ? Do I understand it correctly ?

@paweldomas this `corePoolSize` which is set to `0` means the number of threads in pool to keep even when there is no work to execute. So, since it is a library, I decided that their users don't want to pay some threads when library is not actively using. I configured executor to not too keep "warm" (core) threads if there is no more for more than 10 seconds.


So basically, when pool is created it's empty.
When some task is queued - thread is allocated (if there is not any).
Work is executed.
Thread has no more work for 10 seconds.
Thread is released from pool.
Pool is empty again, because no work to execute.